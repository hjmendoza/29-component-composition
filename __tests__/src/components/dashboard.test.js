import React from "react";
import renderer from "react-test-renderer";
import 'jest-enzyme';
// import {Enzyme} from 'enzyme';
import { configure } from "enzyme";
import { shallow } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

import Dashboard from '../../../src/components/dashboard/dashboard.js';
import NoteCreateForm from '../../../src/components/notecreateform/notecreateform.js';
import NoteList from "../../../src/components/note-list/note-list.js";


describe("Dashboard", () => {
  it("should render without crashing", () => {
    shallow(<Dashboard />);
  });

  it("should have form", () => {
    const wrapper = shallow(<Dashboard />);
    expect(wrapper.find(NoteCreateForm)).toExist();
    expect(wrapper.find(NoteList)).toExist();
  });
});