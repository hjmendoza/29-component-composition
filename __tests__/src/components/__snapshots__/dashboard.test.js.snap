// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard/> (Snapshot Test) renders right 1`] = `
Array [
  <h1>
    Notes
  </h1>,
  <form
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <input
      name="title"
      placeholder="title"
      value=""
    />
    <textarea
      name="content"
      placeholder="content"
      value=""
    />
    <label>
      <span>
        editing
      </span>
      <input
        name="editing"
        type="checkbox"
      />
    </label>
    <label>
      <span>
        completed
      </span>
      <input
        name="completed"
        type="checkbox"
      />
    </label>
    <button
      type="submit"
    >
      Create Note
    </button>
  </form>,
  <ul />,
]
`;

exports[`<Dashboard/> Snapshot Test renders correctly 1`] = `
Array [
  <h1>
    Notes
  </h1>,
  <form
    onChange={[Function]}
    onSubmit={[Function]}
  >
    <input
      name="title"
      placeholder="title"
      value=""
    />
    <textarea
      name="content"
      placeholder="content"
      value=""
    />
    <label>
      <span>
        editing
      </span>
      <input
        name="editing"
        type="checkbox"
      />
    </label>
    <label>
      <span>
        completed
      </span>
      <input
        name="completed"
        type="checkbox"
      />
    </label>
    <button
      type="submit"
    >
      Create Note
    </button>
  </form>,
  <ul />,
]
`;

exports[`<NoteCreateForm/> (Snapshot Test) renders right 1`] = `
<form
  onChange={[Function]}
  onSubmit={[Function]}
>
  <input
    name="title"
    placeholder="title"
    value=""
  />
  <textarea
    name="content"
    placeholder="content"
    value=""
  />
  <label>
    <span>
      editing
    </span>
    <input
      name="editing"
      type="checkbox"
    />
  </label>
  <label>
    <span>
      completed
    </span>
    <input
      name="completed"
      type="checkbox"
    />
  </label>
  <button
    type="submit"
  >
    Create Note
  </button>
</form>
`;
